x-pack/plugin/core/src/main/resources/ecs-template.yml

```diff
@@ -0,0 +1,222 @@
+          match_mapping_type: double
+          runtime:
+            type: double
+
+      # Matching all boolean fields
+      # This matches the following fields: exists
+      # No path matching is needed as it is assumed these fields are already in the json doc as boolean
+      # Elasticsearch would do this mapping automatically but it is specified here for documentation purpose
+      # DISCUSS: If a field "true" as string comes in, will it still be mapped correctly?
+      - strings_as_boolean:
+          match_mapping_type: boolean
+          runtime:
+            type: boolean
+
+      # Matching all string fields by default to keyword
+      # TODO: Does first / last match win in the list?
+      - strings_as_keyword:
+          match_mapping_type: string
+          runtime:
+            type: keyword
+            # Do we need this for runtime?
+            ignore_above: 1024
+
+      # Any message field except the top level one is matched as text but not indexed
+      # TODO: Does this work? Which one has priority?
+      - match_message:
+          path_match: message
+          runtime:
+            type: match_only_text
+
+      # Match all ip fields to ip
+      - match_ip:
+          match_mapping_type: string
+          # DISCUSS: If match could be an array, these mappings could be written much more compact
+          match: ip
+          runtime:
+            type: ip
+
+      # Match all forwarded_ip fields to ip. This is for network.forwarded_ip
+      - match_ip:
+          match_mapping_type: string
+          match: forwarded_ip
+          runtime:
+            type: ip
+
+      # Match all labels fields. Flattened type instead of keyword is used to prevent too many fields to be created.
+      # DISCUSS: Is this the right choice?
+      - match_labels:
+          match_mapping_type: string
+          # TODO: Does this also match the top level labels?
+          match: labels
+          runtime:
+            type: flattened
+
+      # Match location fields to geopoint
+      - match_location:
+          # TODO: What is the correct matching type here?
+          match_mapping_type: object
+          match: location
+          runtime:
+            type: geo_point
+
+      # Match location fields to geopoint
+      - match_location:
+          # TODO: What is the correct matching type here?
+          match_mapping_type: string
+          match: name
+          runtime:
+            # TODO: This is a field type that does not exist today. What ECS does is that for all name fields, it
+            # has a subfield .text to also do text indexing
+            type: text_keyword
+
+      # Match strings fields to wildcard
+      # This matches the following fields: strings
+      # An example field is registry.data.strings
+      - match_strings:
+          # TODO: What is the correct matching type here?
+          match_mapping_type: string
+          match: strings
+          runtime:
+            type: wildcard
```

Comment:
wildcard is also not supported by runtime fields, this would be keyword already with dynamic:runtime, unless you want it indexed.

Reply:
Are there any plans to support wildcard?

A more general thought: There are cases like wildcard or text where runtime field will be less powerful then the indexed variant which I would argue is kind of expected. Would it be possible that instead of the person implementing the template having to think this through, ES would internally do a fallback? Lets say wildcard is internally just treated like a keyword field if used as runtime and if the user sends a query with `*` inside it might return a warning. The reason for the above is that with this we keep the nice property of getting from runtime to index is just moving around the definition instead of having to also select a different type.